commands/audit/sca/pnpm/pnpm.go

```diff
@@ -0,0 +1,168 @@
+	lsArgs := append([]string{"--depth", "Infinity", "--json", "--long"}, params.Args()...)
+	npmLsCmdContent, err := getPnpmCmd(executablePath, workingDir, "ls", lsArgs...).RunWithOutput()
+	if err != nil {
+		return
+	}
+	log.Debug("Pnpm ls command output:\n", string(npmLsCmdContent))
+	output := &[]pnpmLsProject{}
+	if err = json.Unmarshal(npmLsCmdContent, output); err != nil {
+		return
+	}
+	dependencyTrees, uniqueDeps = parsePnpmLSContent(*output)
+	return
+}
+
+func parsePnpmLSContent(projectInfo []pnpmLsProject) (dependencyTrees []*xrayUtils.GraphNode, uniqueDeps []string) {
+	uniqueDepsSet := datastructures.MakeSet[string]()
+	for _, project := range projectInfo {
+		// Parse the dependencies into Xray dependency tree format
+		dependencyTree, uniqueProjectDeps := coreXray.BuildXrayDependencyTree(createProjectDependenciesTree(project), getDependencyId(project.Name, project.Version))
+		// Add results
+		dependencyTrees = append(dependencyTrees, dependencyTree)
+		uniqueDepsSet.AddElements(maps.Keys(uniqueProjectDeps)...)
+	}
+	uniqueDeps = uniqueDepsSet.ToSlice()
+	return
+}
+
+func createProjectDependenciesTree(project pnpmLsProject) map[string]coreXray.DepTreeNode {
+	treeMap := make(map[string]coreXray.DepTreeNode)
+	directDependencies := []string{}
+	// Handle production-dependencies
+	for depName, dependency := range project.Dependencies {
+		directDependency := getDependencyId(depName, dependency.Version)
+		directDependencies = append(directDependencies, directDependency)
+		appendTransitiveDependencies(directDependency, dependency.Dependencies, treeMap)
+	}
+	// Handle dev-dependencies
+	for depName, dependency := range project.DevDependencies {
+		directDependency := getDependencyId(depName, dependency.Version)
+		directDependencies = append(directDependencies, directDependency)
+		appendTransitiveDependencies(directDependency, dependency.Dependencies, treeMap)
+	}
+	if len(directDependencies) > 0 {
+		treeMap[getDependencyId(project.Name, project.Version)] = coreXray.DepTreeNode{Children: directDependencies}
+	}
+	return treeMap
+}
+
+// Return npm://<name>:<version> of a dependency
+func getDependencyId(depName, version string) string {
+	return utils.NpmPackageTypeIdentifier + depName + ":" + version
+}
+
+func appendTransitiveDependencies(parent string, dependencies map[string]pnpmLsDependency, result map[string]coreXray.DepTreeNode) {
+	for depName, dependency := range dependencies {
+		dependencyId := getDependencyId(depName, dependency.Version)
+		if node, ok := result[parent]; ok {
+			node.Children = appendUniqueChild(node.Children, dependencyId)
+		} else {
+			result[parent] = coreXray.DepTreeNode{Children: []string{dependencyId}}
+		}
+		appendTransitiveDependencies(dependencyId, dependency.Dependencies, result)
+	}
+}
+
+func appendUniqueChild(children []string, candidateDependency string) []string {
+	for _, existingChild := range children {
+		if existingChild == candidateDependency {
+			return children
+		}
+	}
```

Comment:
```suggestion
	if slices.Contains(children, candidateDependency) {
		return children
	}
```

You can use "golang.org/x/exp/slices" as long as this project is still Go 1.20.

Reply:
It's still 1.20, when we will update the Go we will handle the issues